modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java

```diff
@@ -135,44 +134,26 @@ public Event process(Event event) throws MuleException {
       ComponentIdentifier operationIdentifier =
           new ComponentIdentifier.Builder().withName(operationModel.getName())
               .withNamespace(extensionModel.getName().toLowerCase()).build();
-      Map<String, Object> originalParameterMap = this.resolverSet.resolve(event).asMap();
-      PolicyProvider policyProvider = policyManager.lookupPolicyProvider();
-      Optional<Policy> policy =
-          policyProvider.findOperationPolicy(event.getContext().getId(), operationIdentifier);
-      if (policy.isPresent()) {
-        Optional<PolicyOperationParametersTransformer> policyOperationParametersTransformer =
+      Optional<OperationPolicy> policy = policyManager.findOperationPolicy(event.getContext().getId(), operationIdentifier);
+      NextOperation nextOperation = operationCallEvent -> {
+        Map<String, Object> parametersMap;
+        Optional<OperationPolicyParametersTransformer> operationPolicyParametersTransformer =
             policyManager.lookupOperationParametersTransformer(operationIdentifier);
-        Event policyEvent;
-        if (policyOperationParametersTransformer.isPresent()) {
-          Message message = policyOperationParametersTransformer.get().fromParametersToMessage(originalParameterMap);
-          policyEvent = Event.builder(event).message((InternalMessage) message).build();
+        if (policy.isPresent() && operationPolicyParametersTransformer.isPresent()) {
+          parametersMap = new HashMap<>();
+          parametersMap
+              .putAll(operationPolicyParametersTransformer.get().fromMessageToParameters(operationCallEvent.getMessage()));
```

Comment:
the createExecutionContext() method below will end up creating the parameters map anyway. Can't that be leveraged? It seems to me like this will cause the parameters to be evaluated twice